### ISSUE TITLE
Failure when executing generateDependencyGraph task

### ISSUE THREAD

**Issue by BlueIce** [vanniktech/gradle-dependency-graph-generator-plugin#13]
When I run `./gradlew generateDependencyGraph` in a project the build is failing if I use version 0.2.0 (see stacktrace). Switching to version 0.1.0 solves the problem. This seems somehow related to the command `command`.

Environment:
```
./gradlew --version                                                                                                                                                                                                        

------------------------------------------------------------
Gradle 4.6
------------------------------------------------------------

Build time:   2018-02-28 13:36:36 UTC
Revision:     8fa6ce7945b640e6168488e4417f9bb96e4ab46c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          1.8.0_151 (Oracle Corporation 25.151-b12)
OS:           Linux 4.9.0-6-amd64 amd64
```

<details>
  <summary>Stacktrace (click to expand)</summary>

```
FAILURE: Build failed with an exception.
 
 * What went wrong:
 Execution failed for task ':generateDependencyGraph'.
 > A problem occurred starting process 'command 'command''
 
 * Try:
  Run with --scan to get full insights.

 
 * Exception is:
 org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':generateDependencyGraph'.
   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:103)
   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
   at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
   at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
   at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
   at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
   at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
   at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
   at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
   at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
   at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
   at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
   at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
   at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:256)
   at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
   at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
   at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
   at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:249)
   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:238)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
   at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
   at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:597)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
   at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
   at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
   at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
 Caused by: org.gradle.process.internal.ExecException: A problem occurred starting process 'command 'command''
   at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:226)
   at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:204)
   at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:349)
   at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:85)
   at org.gradle.internal.operations.BuildOperationIdentifierPreservingRunnable.run(BuildOperationIdentifierPreservingRunnable.java:39)
   ... 3 more
 Caused by: net.rubygrapefruit.platform.NativeException: Could not start 'command'
   at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27)
   at net.rubygrapefruit.platform.internal.WrapperProcessLauncher.start(WrapperProcessLauncher.java:36)
   at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:67)
   ... 4 more
 Caused by: java.io.IOException: Cannot run program "command" (in directory "/ramtmp/gradle_test"): error=2, No such file or directory
   at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
   ... 6 more
 Caused by: java.io.IOException: error=2, No such file or directory
   ... 7 more
```
</details>

**Comment by vanniktech** [comment]
Can you try and check whether the `command` command exists? I'm executing it too see whether you installed graphviz dot or not. If you haven't I'm printing instructions on how to install it.

**Comment by BlueIce** [comment]
In the shell it is working:
```
$ command -v dot
/usr/bin/dot
```
But it is a built-in command:
```
$ which command
command: shell built-in command
```

Do I have to install an additional package?

**Comment by vanniktech** [comment]
That looks all fine. I'm getting the same when running `which command`. Do you happen to know any way to check if a command is executable from Gradle? The `command` seemed like the only solution to me.

**Comment by BlueIce** [comment]
Hm, since `command` is shell built-in it seems kind of logic that it doesn't work (which of course raises the question why is it working on your system).

A workaround may be something like
```
it.executable = "sh"
it.args("-c", "'command -v dot'")
```
though I'm not a gradle expert.

**Comment by vanniktech** [comment]
This doesn't seem to work in my case. Does oh my zsh do something? I'm using it on both Linux and Mac.

**Comment by BlueIce** [comment]
This question seems to be related to my problem: https://stackoverflow.com/questions/46461898/using-gradle-how-can-i-ensure-that-a-command-exists.

**Comment by BlueIce** [comment]
Could you try my example from above without the single quotes?

```
it.executable = "sh"
it.args("-c", "command -v dot")
```

**Comment by vanniktech** [comment]
I'll do that. And I'm also working on integration tests.

**Comment by vanniktech** [comment]
Okay that works. Just tried it on my Ubuntu machine. #14 will fix this. Thanks for all of your help.

**Comment by BlueIce** [comment]
Nice. I'll re-test when the next release is published.

**Comment by vanniktech** [comment]
You can already try out the snapshot and report whether it works or not:

```
buildscript {
  repositories {
    maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
  }
  dependencies {
    classpath "com.vanniktech:gradle-dependency-graph-generator-plugin:0.3.0-SNAPSHOT"
  }
}

apply plugin: "com.vanniktech.dependency.graph.generator"
```

**Comment by BlueIce** [comment]
Ah yes, forgot you are publishing snapshots.

It's working great now, thank you. :smiley:

**Comment by tinchou** [comment]
Can confirm it works

**Comment by vanniktech** [comment]
I just released version 0.3.0 which should be available soon.
